packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -317,7 +318,32 @@ export class SummaryManager extends TypedEventEmitter<ISummaryManagerEvents> imp
         }
 
         if (delayMs > 0) {
-            await delay(delayMs);
+            // eslint-disable-next-line prefer-const
+            let timer;
+            let resolveFn;
+            // Create a listener that will break the delay if we've exceeded the initial delay ops count.
+            const opsListenerFn = () => {
+                if (this.summaryCollection.opsSinceLastAck >= this.opsToBypassInitialDelay) {
+                    clearTimeout(timer);
+                    resolveFn();
+                }
+            };
+            // Create a Promise and capture its resolve function.
+            const delayPromise = new Promise<void>((resolve) => { resolveFn = resolve; }).catch(() => {
+                this.summaryCollection.removeOpListener(opsListenerFn);
+                if (timer) {
+                    clearTimeout(timer);
+                }
+            });
+            // Start a timer to delay creating the summarizer.
+            timer =
+                setTimeout(() => {
+                    resolveFn();
+                },
+                delayMs);
+            this.summaryCollection.addOpListener(opsListenerFn);
+            await delayPromise;
+            this.summaryCollection.removeOpListener(opsListenerFn);
```

Comment:
I think usage of two independent promises (one for timeout and one for op condition) and composition via Promise.race() would make code a bit more readable.

Reply:
That's how I had it at first and decided to get rid of the minor overhead of the second promise + race. But I'm equally happy to go back to that.